platforms/hyperledger-fabric/configuration/roles/create/genesis/tasks/main.yaml

```diff
@@ -0,0 +1,46 @@
+############################################################################################
+- name: Remove old genesis block
+  file:
+    path: "{{ build_path }}/channel-artifacts/genesis.block"
+    state: absent
+
+# This task creates the genesis block by consuming the configtx.yaml file
+- name: "Create genesis block"
+  shell: |
+    cd {{ build_path }}
+    {% if '2.' in network.version %}
+    ./configtxgen -profile {{ genesis.name }} -channelID syschannel  -outputBlock ./channel-artifacts/genesis.block
+    {% else %}
+    ./configtxgen -profile {{ genesis.name }} -outputBlock ./channel-artifacts/genesis.block
+    {% endif %}
+    cat ./channel-artifacts/genesis.block | base64 > ./channel-artifacts/genesis.block.base64
+  when: add_new_org == 'false'
+  tags:
+    - molecule-idempotence-notest
+
+- name: "Check if genesis block has already been written to Vault"
+  shell: |
+    vault kv get -field=genesisBlock secret/crypto/ordererOrganizations
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: vault_genesis_result
+  ignore_errors: yes
+  changed_when: false
+  loop: "{{ network['organizations'] }}"
+  loop_control:
+    loop_var: org
+  when: org.type == "orderer"
+
+- name: "Write genesis block to Vault"  
+  shell: |
+    vault write secret/crypto/ordererOrganizations genesisBlock="$(cat {{build_path}}/channel-artifacts/genesis.block.base64)"
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  tags:
+    - molecule-idempotence-notest
+  loop: "{{ network['organizations'] }}"
+  loop_control:
+    loop_var: org
+  when: add_new_org == 'false' and vault_genesis_result.failed and org.type == "orderer"
```

Comment:
newline?
Also, I think the genesis block should be always written, because in this case if the genesis block is updated because a new channel was added, then that would not be written to Vault

Reply:
Newline added in commit [63ec6fa](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/1053/commits/63ec6fab1b303f01b543e81b6c9e0c98a90154c9). The check was discussed with other devs, and ultimately removed in commit [263350e](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/1053/commits/263350e402b7e7d5ede1115e8e8200d62272e734)